lib/saluki-io/src/net/util/hyper.rs

```diff
@@ -0,0 +1,59 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use pin_project_lite::pin_project;
+use tower::{util::Oneshot, ServiceExt};
+
+/// A Tower service converted into a Hyper service.
+#[derive(Debug, Copy, Clone)]
+pub struct TowerToHyperService<S> {
```

Comment:
We're doing this here because `hyper-util` hasn't caught up to the release of `tower` 0.5, which we're using for the updated retry layer bits.